test/sw_mgmt_spec.rb

```diff
@@ -46,6 +46,24 @@
     }
   end
 
+  describe ".init" do
+    it "initializes package management and returns true" do
+      expect(Yast::PackageLock).to receive(:Connect).with(false).and_return("connected" => true)
+
+      expect(Yast::PackageCallbacks).to receive(:InitPackageCallbacks)
+      expect(Yast::Pkg).to receive(:TargetInitialize)
+      expect(Yast::Pkg).to receive(:TargetLoad)
+      expect(Yast::Pkg).to receive(:SourceRestore).and_return(true)
+
+      expect(Registration::SwMgmt.init).to eq(true)
+    end
+
+    it "returns false when cannot obtain libzypp lock" do
+      expect(Yast::PackageLock).to receive(:Connect).with(false).and_return("connected" => false)
```

Comment:
I would expect this first ```expect``` to be an ```allow```. :-)

Another thing I would probably do, although is open to discussion because it can be considered overkill and too nested:

```ruby
describe ".init" do
  before do
    allow(Yast::PackageLock).to receive(:Connect).and_return("connected" => connected)
  end

  context "when the libzypp lock can be obtained" do
    let(:connected) { true }

    it "returns the result of Pkg::SourceRestore" do
      # Mock dangerous calls
      allow(Yast::PackageCallbacks).to receive(:InitPackageCallbacks)
      allow(Yast::Pkg).to receive(:TargetInitialize)
      allow(Yast::Pkg).to receive(:TargetLoad)

      expect(Yast::Pkg).to receive(:SourceRestore).and_return("something")
      expect(Registration::SwMgmt.init).to eq "something"
    end
  end

  context "when the libzypp lock cannot be obtained" do
    let(:connected) { false }

    it "returns false" do
       expect(Registration::SwMgmt.init).to eq false
    end
  end
end
```

For me, it makes more clear what you are testing and it's not just a copy&paste of the method implementation with an "expect" for each line.